nook/functions/doc_translator/translation_orchestrator.py

```
@@ -0,0 +1,431 @@
+        if preserve_context:
+            self.enable_context_preservation()
+
+    def execute_translation_workflow(self) -> TranslationProject:
+        """Execute complete translation workflow."""
+        start_time = time.time()
+
+        try:
+            # Discover documents
+            documents = self._discover_documents()
+            if not documents:
+                raise ValueError("No documents found to translate")
```

Ruff (0.12.2)

Abstract `raise` to an inner function

(TRY301)